internal/validitywindow/client_test.go

```diff
@@ -238,17 +229,28 @@ func TestBlockFetcherClient_FetchBlocksChangeOfTimestamp(t *testing.T) {
 		}
 	}()
 
-	resultChan := fetcher.FetchBlocks(ctx, tip, &minTimestamp)
+	fetchCtx, cancel := context.WithTimeout(ctx, timeout)
+	defer cancel()
+	resultChan := fetcher.FetchBlocks(fetchCtx, tip, &minTimestamp)
 	receivedBlocks := make(map[uint64]ExecutionBlock[container])
 
 	recvCount := 0
-	for blk := range resultChan {
-		receivedBlocks[blk.GetHeight()] = blk
+loop:
```

Comment:
(not necessary for this PR) - can we re-use `receive` here with a small modification since this is near duplicate code?